lib/tools.js

```diff
@@ -330,92 +359,131 @@ function uuid(givenMac, callback) {
+                    _objects.getObject('system.adapter.vis.0', (err, obj) => {
+                        if (!obj || !obj.native || !obj.native.license) {
+                            // generate new UUID
+                            updateUuid('',  _objects, _uuid => resolve(_uuid));
+                        } else {
+                            // decode obj.native.license
+                            let data;
+                            try {
+                                data = jwt.decode(obj.native.license);
+                            } catch (e) {
 
-                // Add vendor prefix to UUID
-                if (fs.existsSync(VENDOR_FILE)) {
-                    try {
-                        const vendor = require(VENDOR_FILE);
-                        if (vendor.vendor && vendor.vendor.uuidPrefix && vendor.vendor.uuidPrefix.length === 2) {
-                            _uuid = vendor.vendor.uuidPrefix + _uuid;
-                        }
-                    } catch {
-                        console.error(`Cannot parse ${VENDOR_FILE}`);
-                    }
-                }
+                            }
 
-                _objects.setObject('system.meta.uuid', {
-                    type: 'meta',
-                    common: {
-                        name: 'uuid',
-                        type: 'uuid'
-                    },
-                    ts: new Date().getTime(),
-                    from: 'system.host.' + getHostName() + '.tools',
-                    native: {
-                        uuid: _uuid
-                    }
-                }, err => {
-                    if (err) {
-                        console.error('object system.meta.uuid cannot be updated: ' + err);
-                        !--tasks && callback && callback(_uuid);
-                    } else {
-                        _objects.getObject('system.meta.uuid', (err, obj) => {
-                            if (obj.native.uuid !== _uuid) {
-                                console.error('object system.meta.uuid cannot be updated: write protected');
+                            if (!data || !data.uuid) {
+                                // generate new UUID
+                                updateUuid('', _objects, __uuid => resolve(__uuid));
                             } else {
-                                console.log('object system.meta.uuid created: ' + _uuid);
+                                if (data.uuid !== 'ab265f4a-67f9-a46a-c0b2-61e4b95cefe5') {
+                                    updateUuid('', _objects, _uuid => resolve(_uuid));
+                                } else if (data.correct) {
+                                    updateUuid(data.uuid, _objects, _uuid => resolve(_uuid));
+                                } else {
+                                    // Show error
+                                    console.warn(`Your iobroker.vis license must be updated. Please contact info@iobroker.net to get a new license!`);
```

Comment:
I'm not sure anyone will see this console log ...

Reply:
No one must see it. So ignore